### ISSUE TITLE
Excessive script downloads caused by name-matching function

### ISSUE THREAD

**Issue by zhangcandrew** [weecology/retriever#1063]
Right now there is a function called name_matches in engine_tools which helps users download scripts that are closely related to what they type in. I.e. if a user accidentally types `retriever install csv iriis`, the name_matches function should still download iris since it recognizes the user just made a simple error. 

The problem we have right now is that if we type `retriever install csv mapped-plant-quads-co` and only want mapped-plant-quads-co, retriever installs mapped-plant-quads-co, mapped-plant-quads-ks, mapped-plant-quads-id, and mapped-plant-quads-mt due to the name matching function.

**Comment by ethanwhite** [comment]
My personal take is that we should match other package managers and:

1. Only install if the names match exactly
2. If the name doesn't match an existing dataset then check to see if there is something close using `name_matches`, tell the user that there is no dataset by that name, and return a "did you mean XXXX" message.

**Comment by ss-is-master-chief** [comment]
@ethanwhite @zhangcandrew Rather than having different packaged functions like `install_csv`, `install_json`, etc. would it be scalable to have just `install` as a command for the CLI, followed by the type? Something like this : 
```
install --+
          |--- csv
          |--- json
          |--- xml
          |--- msaccess
          |--- postgres
          |--- sqlite
          |--- mysql
```

**Comment by ethanwhite** [comment]
@ss-is-master-chief - I don't understand the relationship between the install functions and the name matching in this issue, so this may be best discussed in a separate issue.

**Comment by ethanwhite** [comment]
Fixed in 8ad74192ce9ca1508452ff3dd6c1dfcf43cafab2
